File: recyclermvp/src/main/java/com/kirchhoff/recyclermvp/MvpRecyclerViewActivity.java
package com.kirchhoff.recyclermvp;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

public class MvpRecyclerViewActivity extends AppCompatActivity {

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.a_mvp_recycler_view);


		RecyclerView recyclerView = findViewById(R.id.recyclerView);
		recyclerView.setLayoutManager(new LinearLayoutManager(this));

		RepositoriesRecyclerAdapter adapter =
				new RepositoriesRecyclerAdapter(new RepositoriesListPresenter());
		recyclerView.setAdapter(adapter);
	}
}

File: recyclermvp/src/main/java/com/kirchhoff/recyclermvp/MvpRecyclerViewActivity.kt
package com.kirchhoff.recyclermvp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MvpRecyclerViewActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.a_mvp_recycler_view)

        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)

        val adapter = RepositoriesRecyclerAdapter(RepositoriesListPresenter())
        recyclerView.adapter = adapter
    }
}
